spec/unit/provider/package/apt_spec.rb

```diff
@@ -86,6 +86,34 @@
     provider.run_preseed
   end
 
+  context "When listing all instances" do
+    let(:args) { ['-W', '--showformat', %Q{'${Status} ${Package} ${Version}\\n'}] }
+    let(:bash_version) { '4.2-5ubuntu3' }
+    let(:bash_installed_output) { "install ok installed bash #{bash_version}\n" }
+    let(:bash_installed_io) { StringIO.new(bash_installed_output) }
+    let(:execpipe_args) { args.unshift('myquery') }
+
+    before do
+      allow(described_class).to receive(:command).with(:dpkgquery).and_return('myquery')
+      allow(described_class).to receive(:command).with(:aptmark, 'showmanual').and_return('bash\n')
```

Comment:
This test if failing because aptmark is defined using `commands :aptmark => "/usr/bin/apt-mark"` which creates an instance and a class method `aptmark` for AptProvider, so the stub here should be for `aptmark` not `command`, Also you need to wrap the string between `""` if you want to use "\n"

```ruby
allow(described_class).to receive(:aptmark).and_return("bash\n")
```

Reply:
@gimmyxd thanks for your help, have copied you suggested code however im confused about this bit.   should this not be 
```ruby
allow(described_class).to receive(:aptmark).with(:showmanual).and_return("bash\n")
```
because of this https://github.com/puppetlabs/puppet/pull/7802/files#diff-361221adf863db9a7cb8c40e8353383cR36

thanks

Reply:
Yes, missed that part, usually you want also to match the parameters that e method is called with.
The allow that i wrote will let aptmark to be called with any paramater, which is not the desired case here.

Reply:
thanks that makes much more sense now